frontend/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import { ThemeProvider } from "styled-components";
+import { getTheme, ITheme } from "./Theme";
+
+interface IThemeContext {
+  dark: boolean;
+  onToggle?: () => void;
+}
+
+const ThemeContext = React.createContext<IThemeContext>({
+  dark: false,
+});
+
+const useTheme = (): IThemeContext => React.useContext(ThemeContext);
```

Comment:
Does useContext return anything? Why we got a wrapper around this? won't useTheme = React.useContext(context) be fine? Also does useContext accept any generics? like useContext<IThemeContext>(context) ?

Reply:
yes it return the current context on the ThemeContext. That wont be fine. It is a custom hook which will be called inside of a react component. May b it works but functions are better if i need to change the theme in a middle of something later.

Reply:
Sure, when you've finished with out messages I'll try just store it directly. I don't think it needs wrapping in a function lol but we see XD

Reply:
lol.. I'm not storing it. I'm fetching it. You cannot make use of a vars here instead of a function. If you make a variable instead of a hook you need to update that var when ever the context updates.